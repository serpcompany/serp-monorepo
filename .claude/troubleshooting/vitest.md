https://vitest.dev/guide/common-errors.html