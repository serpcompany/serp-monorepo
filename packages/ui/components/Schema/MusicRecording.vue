<script setup lang="ts">
  import type { Recording } from '@serp/types/types';
  const props = withDefaults(
    defineProps<{
      song: Recording;
    }>(),
    {
      song: {} as Recording
    }
  );

  // Define a computed schema that dynamically generates JSON-LD based on props
  const schemaData = computed(() => ({
    '@context': 'https://schema.org',
    '@type': 'MusicRecording',
    '@id': props.song.slug,
    name: props.song.name,
    byArtist: {
      '@type': 'MusicGroup',
      name: props.song.artists?.[0]?.name
    },
    duration: props.song['length'],
    inAlbum: props.song.releaseGroup?.name
      ? { '@type': 'MusicAlbum', name: props.song.releaseGroup.name }
      : undefined,
    // isrcCode: props.isrcCode,
    about: {
      '@type': 'CreativeWork',
      name: props.song.name
    },
    // copyrightYear: props.copyrightYear,
    // countryOfOrigin: props.countryOfOrigin,
    //genre: props.genre,
    // keywords: props.keywords,
    // publisher: props.publisher
    //   ? { '@type': 'Organization', name: props.publisher }
    //   : undefined,
    // video: props.video
    //   ? { '@type': 'VideoObject', contentUrl: props.video }
    //   : undefined,
    identifier: props.song.slug,
    sameAs: `https://musicbrainz.org/recording/${props.song.id}`
  }));

  // Apply the schema to the component using useSchemaOrg
  useSchemaOrg(schemaData.value);
</script>
