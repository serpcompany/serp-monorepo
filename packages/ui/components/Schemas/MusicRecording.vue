<script setup lang="ts">
  const props = defineProps({
    name: {
      type: String,
      required: true
    },
    byArtist: {
      type: String,
      required: true
    },
    duration: {
      type: String,
      default: ''
    },
    inAlbum: {
      type: String,
      default: ''
    },
    isrcCode: {
      type: String,
      default: ''
    },
    about: {
      type: String,
      default: ''
    },
    copyrightYear: {
      type: Number,
      default: new Date().getFullYear()
    },
    countryOfOrigin: {
      type: String,
      default: ''
    },
    genre: {
      type: String,
      default: ''
    },
    keywords: {
      type: String,
      default: ''
    },
    publisher: {
      type: String,
      default: ''
    },
    video: {
      type: String,
      default: ''
    },
    identifier: {
      type: String,
      default: ''
    },
    sameAs: {
      type: Array as () => string[],
      default: () => []
    }
  });

  // Define a computed schema that dynamically generates JSON-LD based on props
  const schemaData = computed(() => ({
    '@context': 'https://schema.org',
    '@type': 'MusicRecording',
    '@id': props.identifier || '',
    name: props.name,
    byArtist: {
      '@type': 'MusicGroup',
      name: props.byArtist
    },
    duration: props.duration,
    inAlbum: props.inAlbum
      ? { '@type': 'MusicAlbum', name: props.inAlbum }
      : undefined,
    isrcCode: props.isrcCode,
    about: props.about,
    copyrightYear: props.copyrightYear,
    countryOfOrigin: props.countryOfOrigin,
    genre: props.genre,
    keywords: props.keywords,
    publisher: props.publisher
      ? { '@type': 'Organization', name: props.publisher }
      : undefined,
    video: props.video
      ? { '@type': 'VideoObject', contentUrl: props.video }
      : undefined,
    identifier: props.identifier,
    sameAs: props.sameAs
  }));

  // Apply the schema to the component using useSchemaOrg
  useSchemaOrg(schemaData.value);
</script>
