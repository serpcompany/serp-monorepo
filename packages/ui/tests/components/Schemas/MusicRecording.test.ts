import { describe, expect, it } from 'vitest';
import MusicRecording from '../../../components/Schemas/MusicRecording.vue';
import ComponentRender from '../../componentRender';

(globalThis as unknown).useSchemaOrg = () => ({
  // no-op for schema injection during tests
});

describe('SchemasMusicRecording Snapshot', () => {
  it.each([
    [
      'with minimal required props',
      {
        props: {
          name: 'Song A',
          byArtist: 'Artist A'
        }
      }
    ],
    [
      'with full props',
      {
        props: {
          name: 'Song B',
          byArtist: 'Artist B',
          duration: 'PT3M45S',
          inAlbum: 'Album B',
          isrcCode: 'US-S1Z-99-12345',
          about: 'A hit song',
          copyrightYear: 2020,
          countryOfOrigin: 'USA',
          genre: 'Pop',
          keywords: 'pop, hit, summer',
          publisher: 'Music Corp',
          video: 'https://example.com/video.mp4',
          identifier: 'song-b',
          sameAs: [
            'https://example.com/song-b',
            'https://music.example.com/song-b'
          ]
        }
      }
    ]
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `MusicRecording ${desc}`,
        options,
        MusicRecording
      );
      expect(html).toMatchSnapshot();
    }
  );
});
