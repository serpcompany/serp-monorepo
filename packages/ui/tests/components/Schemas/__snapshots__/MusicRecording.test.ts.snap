// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SchemasMusicRecording Snapshot > renders with full props correctly 1`] = `""`;

exports[`SchemasMusicRecording Snapshot > renders with minimal required props correctly 1`] = `""`;
